RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSeedProducer.cc

```diff
@@ -0,0 +1,529 @@
+#include "TrackingTools/PatternTools/interface/TrajectorySmoother.h"
+#include "TrackingTools/Records/interface/TransientRecHitRecord.h"
+#include "TrackingTools/TrackFitters/interface/TrajectoryFitter.h"
+#include "TrackingTools/TrajectoryParametrization/interface/GlobalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TMath.h"
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+LowPtGsfElectronSeedProducer::LowPtGsfElectronSeedProducer( const edm::ParameterSet& conf, 
+							    const lowptgsfeleseed::HeavyObjectCache* ) :
+  ecalClusters_{consumes<reco::PFClusterCollection>(conf.getParameter<edm::InputTag>("ecalClusters"))},
+  hcalClusters_{consumes<reco::PFClusterCollection>(conf.getParameter<edm::InputTag>("hcalClusters"))},
+  ebRecHits_{consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("EBRecHits"))},
+  eeRecHits_{consumes<EcalRecHitCollection>(conf.getParameter<edm::InputTag>("EERecHits"))},
+  rho_(consumes<double>(conf.getParameter<edm::InputTag>("rho"))),
+  beamSpot_(consumes<reco::BeamSpot>(conf.getParameter<edm::InputTag>("BeamSpot"))),
+  fitter_(conf.getParameter<std::string>("Fitter")),
+  smoother_(conf.getParameter<std::string>("Smoother")),
+  builder_(conf.getParameter<std::string>("TTRHBuilder")),
+  passThrough_(conf.getParameter<bool>("PassThrough")),
+  usePfTracks_(conf.getParameter<bool>("UsePfTracks")),
+  minPtThreshold_(conf.getParameter<double>("MinPtThreshold")),
+  maxPtThreshold_(conf.getParameter<double>("MaxPtThreshold"))
+{
+  if ( usePfTracks_ ) { pfTracks_ = consumes<reco::PFRecTrackCollection>(conf.getParameter<edm::InputTag>("pfTracks")); }
+  else                { kfTracks_ = consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracks")); }
+  produces<reco::ElectronSeedCollection>();
+  produces<reco::PreIdCollection>();
+  produces<reco::PreIdCollection>("HCAL");
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+LowPtGsfElectronSeedProducer::~LowPtGsfElectronSeedProducer() {}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+void LowPtGsfElectronSeedProducer::beginLuminosityBlock( edm::LuminosityBlock const&, 
+							 edm::EventSetup const& setup ) 
+{
+  setup.get<IdealMagneticFieldRecord>().get(field_);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//
+void LowPtGsfElectronSeedProducer::produce( edm::Event& event, 
+					    const edm::EventSetup& setup ) 
+{
+  
+  // Products
+  auto seeds = std::make_unique<reco::ElectronSeedCollection>();
+  auto ecalPreIds = std::make_unique<reco::PreIdCollection>();
+  auto hcalPreIds = std::make_unique<reco::PreIdCollection>();
+  
+  // KF tracks
+  edm::Handle<reco::TrackCollection> kfTracks;
+  if ( !usePfTracks_ ) { event.getByToken(kfTracks_, kfTracks); }
```

Comment:
I think it would be clearer to move this (and the complement for PF tracks below) under the "main" `if ( usePfTracks_ ) {` on lines 100-117.